src/server/db_slice.cc

```diff
@@ -637,7 +642,7 @@ bool DbSlice::Del(DbIndex db_ind, PrimeIterator it) {
     DbContext cntx{db_ind, GetCurrentTimeMs()};
     doc_del_cb_(key, cntx, it->second);
   }
-
+  bumped_items_.erase(it->first.AsRef());
```

Comment:
what about flush operation? maybe DCHECK there that `bumped_items_` are empty?

Reply:
flushdb ? how is this related?
I am adding items to bumped_items_ when we do find for specific item and remove it when the callback is finished. 
If we perform flushdb there will be no items in bumped_items_ as there is not find op done to none of the keys.
Where do you suggest adding the DCHEKC?

Reply:
I meant to add a dcheck that bumped_items_ are indeed empty.

Reply:
yes I understand that you mean to add DCHEKC(bumped_items_.empty()), but where can I add such a check?
It should be empty when I start a cb and after the cb is finished. When the callback is finished I call the new function OnCbFinish which clears the bumped_items_. There is no funciton that is called at before a callback is invoked so where do you think I do this check?

Reply:
@romange ping here